packages/client/src/public-functions/subscribeToRouteStatus.ts

```diff
@@ -262,24 +262,34 @@ export const executeAndSubscribeToRouteStatus = async ({
 
     while (true) {
       console.log(
-        "polling transaction status",
+        "debug: polling transaction status",
         transaction.txHash,
         transaction.chainId,
         routeId
       );
 
       const updatedRouteDetails = routeDetailsMap.get(routeId ?? "");
+      console.log("debug: updatedRouteDetails", updatedRouteDetails);
       const allRelatedRoutesFinal = updatedRouteDetails?.relatedRoutes?.every(
         (relatedRoute) =>
           relatedRoute && isFinalRouteStatus(relatedRoute as RouteDetails)
       );
 
-      if (isFinalState(transaction) && allRelatedRoutesFinal) {
-        break;
+      if (isFinalState(transaction)) {
+        if (
+          updatedRouteDetails?.relatedRoutes &&
+          updatedRouteDetails?.relatedRoutes?.length > 0
+        ) {
+          if (allRelatedRoutesFinal) {
+            break;
+          }
+        } else {
+          break;
+        }
```

Comment:
something like this would be easier to read
if (
  isFinalState(transaction) &&
  (!updatedRouteDetails?.relatedRoutes?.length || allRelatedRoutesFinal)
) {
  break;
}